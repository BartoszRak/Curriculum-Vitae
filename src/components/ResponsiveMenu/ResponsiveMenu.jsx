import React, { useState } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import {
  IconButton, useTheme, useMediaQuery,
} from '@material-ui/core'
import MenuOpenIcon from '@material-ui/icons/Menu'
import MenuCloseIcon from '@material-ui/icons/Close'

import Button from '~components/Button'

import useStyle from './ResponsiveMenu.style'

export function ResponsiveMenu({ classes: overridingClasses, items, className }) {
  const classes = { ...useStyle(), ...overridingClasses }
  const [toggle, setToggle] = useState(false)
  const theme = useTheme()
  const matchDimensions = useMediaQuery(theme.breakpoints.down('md'))

  return (
    <div className={classNames(classes.root, className)}>
      {matchDimensions && (
        <div className={classes.actions}>
          <IconButton
            data-testid="mobile-menu-open-icon"
            onClick={() => {
              setToggle(!toggle)
            }}
          >
            {!toggle ? (
              <MenuOpenIcon color="secondary" />
            ) : (
              <MenuCloseIcon color="secondary" />
            )}
          </IconButton>
        </div>
      )}
      {(!matchDimensions || toggle) && (
        <div
          aria-label="Navigate to"
          className={classes.buttonsHolder}
          role="navigation"
        >
          {items.map(route => (
            <Button
              classes={{
                label: classes.buttonLabel,
                root: classes.button,
              }}
              color="secondary"
              key={route.name}
              {...route.props}
            >
              {route.name}
            </Button>
          ))}
        </div>
      )}
    </div>
  )
}

ResponsiveMenu.propTypes = {
  className: PropTypes.string,
  classes: PropTypes.object,
  items: PropTypes.arrayOf(
    PropTypes.shape({
      elementId: PropTypes.string,
      name: PropTypes.string,
    })
  ).isRequired,
}

export default ResponsiveMenu
