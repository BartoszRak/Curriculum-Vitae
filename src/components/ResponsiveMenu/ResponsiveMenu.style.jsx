import { makeStyles } from '@material-ui/core'

export default makeStyles(({ breakpoints, palette, spacing }) => ({
  actionButton: {
    color: palette.primary.contrastText,
  },
  actionText: {
    marginLeft: spacing(1),
    borderBottom: `1px solid ${palette.primary.contrastText}`,
  },
  actions: {
    alignItems: 'center',
    display: 'flex',
    height: 64,
    [breakpoints.down('xs')]: {
      height: 'auto',
    },
  },
  button: {
    '&:hover': {
      background: 'rgba(0, 0, 0, 0.05)',
    },
    borderRadius: 100,
    marginRight: spacing(4),
    padding: spacing(0.5, 2),
    [breakpoints.down('md')]: {
      margin: spacing(0.5, 0),
      padding: spacing(1, 2),
    },
  },
  buttonLabel: {
    color: palette.secondary.light,
    fontSize: '1rem',
    fontWeight: 'normal',
    [breakpoints.down('md')]: {
      fontSize: '1.2rem',
    },
    [breakpoints.down('xs')]: {
      fontSize: '1rem',
    },
  },
  buttonsHolder: {
    [breakpoints.down('md')]: {
      display: 'flex',
      flex: '1 1 auto',
      flexDirection: 'column',
      justifyContent: 'center',
      padding: spacing(1, 0),
    },
  },
  root: {
    display: 'flex',
    flex: '1 1 auto',
    flexDirection: 'column',
  },
}))
