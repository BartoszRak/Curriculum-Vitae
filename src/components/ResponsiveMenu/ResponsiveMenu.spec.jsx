import React from 'react'
import { render, cleanup } from '@testing-library/react'

import { ResponsiveMenu } from './ResponsiveMenu'

jest.unmock('@material-ui/core')
jest.unmock('@material-ui/icons')

let routes

describe('component: ResponsiveMenu', () => {
  afterEach(cleanup)

  beforeEach(() => {
    routes = [{
      elementId: '1 test element id',
      name: '1 test name',
    }, {
      elementId: '2 test element id',
      name: '2 test name',
    }]
  })
  describe('rendering', () => {
    test('render without crash ', () => {
      const wrapper = render(<ResponsiveMenu items={routes} />)

      expect(wrapper).toBeTruthy()
    })

    test('match snapshot ', () => {
      const wrapper = render(<ResponsiveMenu items={routes} />)

      expect(wrapper).toMatchSnapshot()
    })

    test('render routesd', () => {
      const { getByText } = render(<ResponsiveMenu items={routes} />)

      routes.forEach(route => {
        const routeEl = getByText(route.name)
        expect(routeEl).toBeVisible()
      })
    })
  })
})
