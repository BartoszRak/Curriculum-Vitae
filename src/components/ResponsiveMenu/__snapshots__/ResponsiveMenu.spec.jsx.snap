// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: ResponsiveMenu rendering match snapshot  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-root-45"
      >
        <div
          aria-label="Navigate to"
          class="makeStyles-buttonsHolder-44"
          role="navigation"
        >
          <button
            class="MuiButtonBase-root MuiButton-root makeStyles-button-42 MuiButton-text MuiButton-textSecondary"
            role="button"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label makeStyles-buttonLabel-43"
            >
              1 test name
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root makeStyles-button-42 MuiButton-text MuiButton-textSecondary"
            role="button"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label makeStyles-buttonLabel-43"
            >
              2 test name
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-root-45"
    >
      <div
        aria-label="Navigate to"
        class="makeStyles-buttonsHolder-44"
        role="navigation"
      >
        <button
          class="MuiButtonBase-root MuiButton-root makeStyles-button-42 MuiButton-text MuiButton-textSecondary"
          role="button"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label makeStyles-buttonLabel-43"
          >
            1 test name
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root makeStyles-button-42 MuiButton-text MuiButton-textSecondary"
          role="button"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label makeStyles-buttonLabel-43"
          >
            2 test name
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
